src/System/include/BipedalLocomotion/System/DynamicalSystem.h

```diff
@@ -0,0 +1,107 @@
+/**
+ * @file DynamicalSystem.h
+ * @authors Giulio Romualdi
+ * @copyright 2020 Istituto Italiano di Tecnologia (IIT). This software may be modified and
+ * distributed under the terms of the GNU Lesser General Public License v2.1 or any later version.
+ */
+
+#ifndef BIPEDAL_LOCOMOTION_SYSTEM_DYNAMICAL_SYSTEM_H
+#define BIPEDAL_LOCOMOTION_SYSTEM_DYNAMICAL_SYSTEM_H
+
+#include <memory>
+#include <tuple>
+
+#include <BipedalLocomotion/GenericContainer/TemplateHelpers.h>
+#include <BipedalLocomotion/ParametersHandler/IParametersHandler.h>
+
+namespace BipedalLocomotion
+{
+namespace System
+{
+
+/**
+ * DynamicalSystem defines a continuous time dynamical system, i.e. \f$\dot{x}=f(x, u, t)\f$. Please
+ * inherit publicly from this class in order to define your custom dynamical system.
+ * @tparam State type used for describing the state (i.e. it has to be a std::tuple vectors/matrices
+ * or in general classes).
+ * @tparam StateDerivative type used for describing the state derivative (i.e. it has to be a
+ * std::tuple of vectors/matrices or in general classes).
+ * @tparam Input type used for describing the input (i.e. it has to be a std::tuple of
+ * vectors/matrices or in general classes).
+ */
+template <typename State, typename StateDerivative, typename Input>
+class DynamicalSystem
+{
+    static_assert(is_specialization<State, std::tuple>::value);
+    static_assert(is_specialization<StateDerivative, std::tuple>::value);
+    static_assert(is_specialization<Input, std::tuple>::value);
```

Comment:
Maybe you can add some message about the error.

Reply:
Explanation improved in 5db82f3b770ba5b6cb6727f4c00aaee3d0dea56d